src/huggingface_hub/hf_api.py

```diff
@@ -5020,6 +5020,79 @@ def delete_space_secret(self, repo_id: str, key: str, *, token: Optional[str] =
         )
         hf_raise_for_status(r)
 
+    @validate_hf_hub_args
+    def get_space_variables(self, repo_id: str, *, token: Optional[str] = None) -> Iterable[SpaceVariable]:
+        """Gets all variables from a Space.
+
+        Variables allow to set environment variables to a Space without hardcoding them.
+        For more details, see https://huggingface.co/docs/hub/spaces-overview#managing-secrets-and-environment-variables
+
+        Args:
+            repo_id (`str`):
+                ID of the repo to query. Example: `"bigcode/in-the-stack"`.
+            token (`str`, *optional*):
+                Hugging Face token. Will default to the locally saved token if not provided.
+        """
+        r = get_session().get(
+            f"{self.endpoint}/api/spaces/{repo_id}/variables",
+            headers=self._build_hf_headers(token=token),
+        )
+        hf_raise_for_status(r)
+        for k, v in r.json().items():
+            yield (SpaceVariable(k, v))
+
+    @validate_hf_hub_args
+    def add_space_variable(
+        self, repo_id: str, key: str, value: str, description: Optional[str] = str(), *, token: Optional[str] = None
```

Comment:
```suggestion
        self, repo_id: str, key: str, value: str, *, description: Optional[str] = None, token: Optional[str] = None
```

Same comment as https://github.com/huggingface/huggingface_hub/pull/1594#discussion_r1295938121. I'd prefer keywork-only parameter + it's best to default to None when the value is not known instead of `""`.